### PR TITLE
Pivot: support custom rendering of pivot links

### PR DESCRIPTION
**Pull request by cschleiden**
#### Pull request checklist

- [x] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
- [x] Documentation update

#### Description of changes

Supports passing in a custom renderer for pivot links. For example, to show additional annotations

image

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Pivot/Pivot.tsx [hunk 1]
```diff
@@ -142,13 +139,37 @@ export class Pivot extends React.Component<IPivotProps, IPivotState> {
...
+
+    let text: JSX.Element;
+    if (linkText !== undefined) {
+      text = <span className='ms-Pivot-text'>{ link.linkText }</span>;
+    }
+
+    return <span className='ms-Pivot-link-content'>
+      { icon }
+      { text }
+      { count }
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Pivot/Pivot.tsx]: can we do these inline 
`{ itemCount && <span className='ms-Pivot-count'>({ itemCount })</span> }` ?. I find it easier to read JSX that way. Not pushing for it though.

File: packages/office-ui-fabric-react/src/components/Pivot/Pivot.tsx [hunk 2]
```diff
@@ -98,6 +99,8 @@ export class Pivot extends React.Component<IPivotProps, IPivotState> {
    * Renders the set of links to route between pivots
    */
   private _renderPivotLinks() {
+    const { } = this.props;
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Pivot/Pivot.tsx]: nit: extra i guess

Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Pivot/Pivot.tsx]: oh, of course.

File: packages/office-ui-fabric-react/src/components/Pivot/Pivot.tsx (hunk 1 above)
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Pivot/Pivot.tsx]: I think your example wouldn't work for an `itemCount` of `0`. We could do 
`{ itemCount !== undefined && <span className='ms-Pivot-count'>({ itemCount })</span> }` , however

**Pull request merged**